packages/math-input/src/components/key-translator.ts

```diff
@@ -0,0 +1,182 @@
+import Key from "../data/keys";
+import {MathQuillInterface} from "../types";
+
+enum ActionType {
+    WRITE = "write",
+    CMD = "cmd",
+    KEYSTROKE = "keystroke",
+    MQ_END = 0,
+}
+
+type MathQuillCallback = (mathQuill: MathQuillInterface) => void;
+
+function getGenericCallback(
+    str: string,
+    type: ActionType = ActionType.WRITE,
+): MathQuillCallback {
+    return function (mathQuill: MathQuillInterface) {
+        switch (type) {
+            case ActionType.WRITE: {
+                mathQuill.write(str);
+                return;
+            }
+            case ActionType.CMD: {
+                mathQuill.cmd(str);
+                return;
+            }
+            case ActionType.KEYSTROKE: {
+                mathQuill.keystroke(str);
+                return;
+            }
+        }
+    };
+}
+
+const keyToMathquillMap: Record<Key, MathQuillCallback | null> = {
+    CDOT: getGenericCallback("\\cdot"),
+    COS: getGenericCallback("cos"),
+    DECIMAL: getGenericCallback("."),
+    DIVIDE: getGenericCallback("\\div"),
+    EQUAL: getGenericCallback("="),
+    EXP: getGenericCallback("^"),
+    EXP_2: getGenericCallback("^2"),
+    EXP_3: getGenericCallback("^3"),
+    GEQ: getGenericCallback("\\geq"),
+    GT: getGenericCallback(">"),
+    LEQ: getGenericCallback("\\leq"),
+    LN: getGenericCallback("\\ln"),
+    LOG: getGenericCallback("\\log"),
+    LOG_N: getGenericCallback("log_{ }"),
+    LT: getGenericCallback("<"),
+    MINUS: getGenericCallback("-"),
+    NEGATIVE: getGenericCallback("-"),
```

Comment:
Do the truly write the same value to the input?

Reply:
It's from MathWrapper:

```
    ["MINUS"]: {str: "-", fn: ActionType.WRITE},
    ["NEGATIVE"]: {str: "-", fn: ActionType.WRITE},
```